### PR TITLE
PMM-4351 Split QAN protocol to two parts.

### PR DESCRIPTION
**Pull request by BupycHuk**

### CODE REVIEW

**Comment by JNKPercona** [comment]
http://docs-test.cd.percona.com/PR-397/

**Comment by JNKPercona** [comment]
http://docs-test.cd.percona.com/PR-397/

File: api/agentpb/collector.proto [hunk 1]
```diff
@@ -0,0 +1,325 @@
...
+    // Type of service
+    string service_type = 12;
+    // Environment name
+    string environment = 13;
+    // Availability zone
+    string az = 14;
+    // Region name
+    string region = 15;
+    // Node model
+    string node_model = 16;
```
Comment (AlekSi) [inline api/agentpb/collector.proto]: Many of those fields are set by pmm-managed, not by pmm-agent, so we can remove them from that half. That's the whole point of this task.

File: api/agentpb/collector.proto [hunk 2]
```diff
@@ -0,0 +1,325 @@
...
+    float m_docs_scanned_sum = 12;
+    float m_docs_scanned_min = 13;
+    float m_docs_scanned_max = 14;
+    float m_docs_scanned_p99 = 15;
+  }
+  // PostgreSQL contains fields only for PostgreSQL.
+  message PostgreSQL {
+    //
+    // PostgreSQL metrics.
+    //
```
Comment (AlekSi) [inline api/agentpb/collector.proto]: Those comments at the start of the messages are superfluous.

File: api/agentpb/collector.proto [hunk 3]
```diff
@@ -0,0 +1,325 @@
...
+    // How many queries was with warnings in bucket.
+    float num_queries_with_warnings = 1;
+    // List of warnings: {code: count}.
+    map<uint64, uint64> warnings = 2;
+    // How many queries was with error in bucket.
+    float num_queries_with_errors = 3;
+    // List of errors: {code: count}.
+    map<uint64, uint64> errors = 4;
+    // Amount queries in this bucket.
+    float num_queries = 5;
```
Comment (AlekSi) [inline api/agentpb/collector.proto]: I'm pretty sure that one is common. :)

File: api/agentpb/collector.proto [hunk 4]
```diff
@@ -0,0 +1,325 @@
...
+    // How many times query_time was found.
+    float m_query_time_cnt = 6;
+    // The statement execution time in seconds.
+    float m_query_time_sum = 7;
```
Comment (askomorokhov) [inline api/agentpb/collector.proto]: I think m_query_time_sum - should be in common message as it is also valid for mongo and postgres.

File: api/agentpb/collector.proto [hunk 5]
```diff
@@ -0,0 +1,325 @@
...
+    //
+    // Dimension Group.
+    //
+    // Service Name (can be several Services on one Node).
+    string service_name = 3;
+    // PostgreSQL database.
+    string database = 4;
+    // MySQL database or PostgreSQL schema.
+    string schema = 5;
+    // client user name.
```
Comment (AlekSi) [inline api/agentpb/collector.proto]: We should add `repeated string tables` there.

File: api/agentpb/collector.proto (hunk 3 above)
Comment (BupycHuk) [reply api/agentpb/collector.proto]: yeah, it's in progress :) I have moved them to common, but forgot to commit it.

**Comment by JNKPercona** [comment]
http://docs-test.cd.percona.com/PR-397/

**Comment by JNKPercona** [comment]
http://docs-test.cd.percona.com/PR-397/

**Comment by JNKPercona** [comment]
http://docs-test.cd.percona.com/PR-397/

**Comment by JNKPercona** [comment]
http://docs-test.cd.percona.com/PR-397/

**Comment by JNKPercona** [comment]
http://docs-test.cd.percona.com/PR-397/

**Comment by JNKPercona** [comment]
http://docs-test.cd.percona.com/PR-397/

File: api/agentpb/agent.proto [hunk 6]
```diff
@@ -37,7 +37,7 @@ message Pong {
 
 // QANCollectRequest is an AgentMessage for sending QAN data for qan-api.
 message QANCollectRequest {
-  repeated MetricsBucket metrics_bucket = 1;
+  CollectRequest request = 1;
```
Comment (AlekSi) [inline api/agentpb/agent.proto]: Uh, why?

**Comment by AlekSi** [comment]
@BupycHuk it is still WIP?

**Comment by JNKPercona** [comment]
http://docs-test.cd.percona.com/PR-397/

**Comment by BupycHuk** [comment]
@AlekSi no, it's not

**Comment by JNKPercona** [comment]
http://docs-test.cd.percona.com/PR-397/

**Comment by JNKPercona** [comment]
http://docs-test.cd.percona.com/PR-397/

**Comment by JNKPercona** [comment]
http://docs-test.cd.percona.com/PR-397/

**Comment by JNKPercona** [comment]
http://docs-test.cd.percona.com/PR-397/

**Pull request merged**